### PR TITLE
update .Readme and modified github actions

### PR DESCRIPTION
**Pull request by sonali8434**
@jwokaty, Please Review My Changes in Readme file.

Before we ask Levi's for the final review, I have made updates to the readme file to provide instructions to users.

- In this pull request, both @jwokaty and I have made changes to the GitHub Actions. These changes allow users to use this package in their forked repository without the need to create a new repository.
- We have also added comments in the auto-generated pull request to provide information about the converted files and to indicate which files need to be converted manually.

### CODE REVIEW

**Comment by lwaldron** [comment]
* What are the advantages of this method (let's call it v2) of forking the package's GitHub repo and then copying this repo's .github directory onto it, as opposed to the original approach (v1) of creating a new repo to work on, then copying the new vignettes onto a fork/PR of the Bioconductor package? My original thoughts in favor of v1 were a) that a number of old packages probably don't even have a GitHub repo, but every Bioconductor package has a repo on git.bioconductor.org, and b) the conversion part could be done entirely through GH / GHA. For packages that exist only on git.bioconductor.org, the user will have to go through some extra steps to convert that to a GitHub repository. 

* Step 11. "After merging the pull request, create a new branch using the command `git switch -c file-Rmd` for the file you wish to create a pull request for" is a bit unclear to me - won't this pull request multiple .Rmd files if there is more than one vignette, plus the .github directory?

* should probably still include the former step 6 of "Check and edit the Rmd vignettes of the new repository until they work" or be even more explicit that the automatically generated Rmd files will still need manual editing.

**Comment by jwokaty** [comment]
> * What are the advantages of this method (let's call it v2) of forking the package's GitHub repo and then copying this repo's .github directory onto it, as opposed to the original approach (v1) of creating a new repo to work on, then copying the new vignettes onto a fork/PR of the Bioconductor package? My original thoughts in favor of v1 were a) that a number of old packages probably don't even have a GitHub repo, but every Bioconductor package has a repo on git.bioconductor.org, and b) the conversion part could be done entirely through GH / GHA. For packages that exist only on git.bioconductor.org, the user will have to go through some extra steps to convert that to a GitHub repository.

@lwaldron I think the main advantage of using v2 is that you don't have to create multiple repositories. You add the action to your forked repository in v2. v1 is nice in that you only need to make a repository using the name of the package you want to convert then it performs the conversion; however, we generally have to make manual edits even if Rnw2Rmd or pandoc successfully convert the vignette, and we do this via a fork of the package's GitHub repo. (If a package doesn't have a GitHub repo, we ask the maintainer to make one.)

One concern I have is the length/complexity of steps involved as our Outreachy contributors know some git but branches, PRs, etc. tend to be confusing. If we want to stick with v1, my suggestion here is that we don't use the same name as the Bioconductor package. I believe workflow_dispatch can take inputs (I have not tested this!), so maybe the conversion template repository gets another name and then you enter the name of the package in the workflow dispatch. This might actually be good then it could be reused for multiple package conversions?

If we continue with v2, we can improve the documentation and it also has some other edits, such as adding a message about whether a vignette was able to be converted and it also removed the section to install R since it's using Bioconductor's Docker.

It might also be good to limit the conversion to a single vignette rather than doing all by setting the workflow dispatch input to the name of the vignette so that we have 1 vignette for 1 PR that contributor can keep adding to.

**Comment by lwaldron** [comment]
I'm certainly willing to defer on what will be simpler for interns. But just to explain some of my thinking, for example:

> Copy and paste the `.GitHub` folder from the conversion template into the cloned repository of the package you want to convert and then commit and push .Github file in the colned repository.

will require some extra explanation since folders starting with `.` are by default hidden in OSX and I think Windows file explorers, and they will also have to install a git client and learn how to commit and push. v1 could be done without ever using git, just by working on your own package created by the action and then using GitHub's "Add File" button on the target repository to create a Pull Request, or by emailing the Rmd file to the package maintainer for the older packages that don't have a GitHub repository (not sure how many that is, but maybe significant for packages still using Sweave?). The v2 instructions look more challenging to me than instructing contributors to run `BiocManager::install("Bioconductor/Rnw2Rmd)` and `Rnw2Rmd::Rnw2Rmd()` on a locally cloned fork. Do we need some novice testers to try both approaches?

Using a different repo name shouldn't be too hard if there are instructions like "add a _ followed by letters and numbers" and the Action stripped that part to figure out which Bioconductor package to clone.

**Comment by sonali8434** [comment]
Thank you, @lwaldron. Your explanations are clear to me. Our main goal is similar: to make the process easier for interns and future contributors. While I have limited knowledge of GitHub, I have tested all the methods and, for me, running `BiocManager::install("Bioconductor/Rnw2Rmd")` and `Rnw2Rmd::Rnw2Rmd()` on a locally cloned fork is the simplest method.

I agree with you that v2 is a bit complicated. However, we have decided to keep v2 as a process because it helps us convert the whole package containing a larger number of .Rnw files at once. Then, we can create different branches from that repository and make pull requests one by one after manually converting them.

After reading your concerns, I completely understand your point. Thank you for guiding us in the right direction. I appreciate your words.

File: .github/workflows/Sweave2rmd.yaml
Comment (jwokaty) [inline .github/workflows/Sweave2rmd.yaml]: I think this should probably be `git rm */vignettes/*.Rnw` otherwise it can remove .Rnw files in other paths.

File: .github/workflows/Sweave2rmd.yaml
Comment (jwokaty) [inline .github/workflows/Sweave2rmd.yaml]: Add back

```
      - name: create repo
        run: |
          echo $REPO_NAME
          git clone "https://git.bioconductor.org/packages/$REPO_NAME"
          rm -rf $REPO_NAME/.git*
          mv $REPO_NAME/* .
          rm -rf $REPO_NAME
```

In the future, this should probably be replaced with a workflow_dispatch input to specify the repository to clone.

File: README.md
Comment (jwokaty) [inline README.md:1]: Remove the changes to this file

**Review by jwokaty — CHANGES_REQUESTED** [review]
These are the changes that should be made to stay with v1. I've made some inline comments to the workflow. I believe we can just undo the changes to the readme file.

File: README.md
Comment (sonali8434) [reply README.md:1]: Done

**Comment by sonali8434** [comment]
@jwokaty I have made all the requested changes please review.

**Review by jwokaty — APPROVED** [review]
@sonali8434 Thanks for these changes.

@lwaldron Could we ask you to look at this PR? This is adjusted back to stay with v1 but adds a workflow dispatch and a message in the PR about  .Rnw files converted to .Rmd and files that failed to convert. The PR also adds ".Rmd" and removes "/vignettes/*.Rnw" files. It removes the set up for r and dependencies since it is using bioconductor_docker.

File: .github/workflows/Sweave2rmd.yaml
Comment (lwaldron) [inline .github/workflows/Sweave2rmd.yaml:1]: This looks great to me! Keeps the user simplicity of v1 while adding useful features to the GHA.

**Pull request merged**
